### PR TITLE
gnome: update to 45.0

### PR DESCRIPTION
**Pull request by chrysos349**
@oreo639 @Gottox 
Your templates are a part of this PR.

I've been using `gnome-45.3` for several weeks now.

**MERGE FIRST:**
- #48560
- #48983 

**NOTES:**
- None of the packages in `gnome-core` require the updated `gjs` or `webkit2gtk`. Packages in `gnome` do require them.
- `GPaste` is not a part of gnome, but the current version in the repo is not compatible with `mutter-45.0`. At the same time, `GPaste-45.0` requires newer `gjs>=1.77.2`.

**TODO:**
- [x] `GPaste-45.0` ( wait till `gjs-1.78.4` gets merged)
- [x] `sysprof-45.2`, a part of #49096 (optional, low-priority)
- [x] #48839 (optional for gnome-core, required for gnome)

#### Testing the changes
- I tested the changes in this PR: **YES**

#### Local build testing
- I built this PR locally for my native architecture, (x86_64)
- I built this PR locally for these architectures (if supported. mark crossbuilds):
  - aarch64-musl x
  - armv7l x
  - armv6l-musl x

### CODE REVIEW

**Comment by chrysos349** [comment]
**TODO**: forgot to add the updated template of `GPaste-45.0` which is compatible with `mutter-45.0`

P.S. I'm not sure if I should've used `ci skip` for this pr.

**Comment by oreo639** [comment]
Some comments.
evolution, evolution-data-sever, and evolution-ews should be updated together and should be a separate PR.
grilo should be a separate PR.
gtk4 should be a separate PR.
at-spi2 can probably be a separate PR.

gnome-core should be 45.0, not 45.4.

**Comment by chrysos349** [comment]
Created separate PRs for select packages.

File: common/shlibs [hunk 1]
```diff
@@ -4284,3 +4284,4 @@ libunicode_ucd.so.0.4 libunicode-0.4.0_1
 libunicode_loader.so.0.4 libunicode-0.4.0_1
 force-stage.so.0.1 void-force-stage-0.1_1
 libliftoff.so.0 libliftoff-0.4.1_1
+libmutter-mtk-13.so.0 mutter-45.0_1
```
Comment (oreo639) [inline common/shlibs:4287]: This should be with the other mutter libs.

**Comment by oreo639** [comment]
You updated gnome-sudoku in the PR, I assume that is an accident?
Feel free to make a separate PR for it though.

**Comment by oreo639** [comment]
Btw, epiphany needs to be revbumped for gcr4. (GPaste as well, although that will be updated ofc)

**Comment by chrysos349** [comment]
`budgie-desktop-10.7.2_3` needs to be updated to be compatible with `mutter-45.*`

```
budgie-desktop-10.7.2_3: broken, unresolvable shlib `libmutter-12.so.0'
budgie-desktop-10.7.2_3: broken, unresolvable shlib `libmutter-clutter-12.so.0'
budgie-desktop-10.7.2_3: broken, unresolvable shlib `libmutter-cogl-12.so.0'
```

**Comment by oreo639** [comment]
Yes, you need to revbump it.

**Comment by chrysos349** [comment]
> Yes, you need to revbump it. (10.7.1+ already supports mutter-12)

unfortunately, it needs to support mutter-13 (which is provided by `mutter-45.*`). revbumping `budgie-desktop` will produce the error
```
Run-time dependency libmutter-12 found: NO (tried pkgconfig and cmake)
Run-time dependency libmutter-11 found: NO (tried pkgconfig and cmake)
Run-time dependency libmutter-10 found: NO (tried pkgconfig)

src/wm/meson.build:45:21: ERROR: Dependency "libmutter-10" not found, tried pkgconfig
```

**Comment by oreo639** [comment]
Yeah, I got mixed up. Budgie will need to be updated to use magpie instead of mutter.

File: srcpkgs/gnome-initial-setup/template [hunk 2]
```diff
@@ -18,4 +18,8 @@ license="GPL-2.0-or-later"
 homepage="https://wiki.gnome.org/Design/OS/InitialSetup"
 changelog="https://gitlab.gnome.org/GNOME/gnome-initial-setup/-/raw/master/NEWS"
 distfiles="${GNOME_SITE}/gnome-initial-setup/${version%%.*}/gnome-initial-setup-${version}.tar.xz"
-checksum=593cfc6dc8f82a98671b94c035b97dbe88ef56e708780b2ad1d2324e4d04bbff
+checksum=363e09aa3308e7f4074e066253a0182b322ab60376743de178bbb400e54ba8ee
+
+if [ "$CROSS_BUILD" ]; then
+	hostmakedepends+=" dconf"
```
Comment (oreo639) [inline srcpkgs/gnome-initial-setup/template:24]: Add this to hostmakedepends unconditionally.

File: srcpkgs/gnome-control-center/template [hunk 3]
```diff
@@ -14,24 +14,28 @@ makedepends="ModemManager-devel accountsservice-devel alsa-plugins
...
+checksum=b1e949c4e86c0624ec6a6ed0df09c9fae2b22983c1dca60ed86aec8ef0b24004
 make_check=no # needs X11 and requires altered filesystem
 
 pre_build() {
 	# q66: racey, included from elsewhere but not dependencies
 	ninja -C build panels/network/cc-network-resources.h
 }
 
+if [ "$CROSS_BUILD" ]; then
+	hostmakedepends+=" setxkbmap"
```
Comment (oreo639) [inline srcpkgs/gnome-control-center/template:36]: Same here.

File: srcpkgs/gnome-bluetooth/template [hunk 4]
```diff
@@ -15,11 +15,15 @@ license="GPL-2.0-or-later, LGPL-2.1-or-later"
...
 changelog="https://gitlab.gnome.org/GNOME/gnome-bluetooth/-/raw/master/NEWS"
 distfiles="${GNOME_SITE}/gnome-bluetooth/${version%.*}/gnome-bluetooth-${version}.tar.xz"
-checksum=a4f5f1ac2dbb7b7b92f7d6d2b42c030f62aed9b5426b9dc116946041f3cba8f7
+checksum=76c241e8ca2c9b1035364535ca26084f89fc5c0e1829510f8909583115fcc2db
 
 build_options="gir"
 build_options_default="gir"
 
+if [ "$CROSS_BUILD" ]; then
+	hostmakedepends+=" libxml2"
```
Comment (oreo639) [inline srcpkgs/gnome-bluetooth/template:24]: Same here

File: srcpkgs/gnome-initial-setup/template (hunk 2 above)
Comment (chrysos349) [reply srcpkgs/gnome-initial-setup/template]: fixed

File: srcpkgs/gnome-control-center/template (hunk 3 above)
Comment (chrysos349) [reply srcpkgs/gnome-control-center/template]: fixed

File: srcpkgs/gnome-bluetooth/template (hunk 4 above)
Comment (chrysos349) [reply srcpkgs/gnome-bluetooth/template]: fixed

File: srcpkgs/zenity/template [hunk 5]
```diff
@@ -1,18 +1,19 @@
...
+configure_args="-Dwebkitgtk=$(vopt_if webkit true false)"
+hostmakedepends="gettext glib-devel gtk-update-icon-cache help2man itstool
+ pkg-config"
+makedepends="$(vopt_if webkit libwebkitgtk60-devel) libadwaita-devel"
 short_desc="Display GNOME dialogs from the command line"
 maintainer="Enno Boland <gottox@voidlinux.org>"
 license="LGPL-2.0-or-later"
 homepage="https://help.gnome.org/users/zenity/"
-changelog="https://gitlab.gnome.org/GNOME/zenity/-/raw/zenity-3-44/NEWS"
+changelog="https://download.gnome.org/sources/zenity/4.0/zenity-4.0.1.news"
```
Comment (oreo639) [inline srcpkgs/zenity/template:14]: This should point to `https://gitlab.gnome.org/GNOME/zenity/-/raw/master/NEWS`

Comment (chrysos349) [reply srcpkgs/zenity/template]: fixed

**Comment by oreo639** [comment]
budgie-desktop has been updated.

**Comment by oreo639** [comment]
folks and gtksourceview4 can be separate PRs.

**Comment by chrysos349** [comment]
added `gnome-remote-desktop`

File: srcpkgs/gdm/template [hunk 6]
```diff
@@ -11,23 +11,24 @@ configure_args="
...
 short_desc="GNOME Display Manager"
 maintainer="Orphaned <orphan@voidlinux.org>"
 license="GPL-2.0-or-later"
 homepage="https://wiki.gnome.org/Projects/GDM"
-changelog="https://gitlab.gnome.org/GNOME/gdm/-/raw/gnome-44/NEWS"
+changelog="https://gitlab.gnome.org/GNOME/gdm/-/raw/gnome-45/NEWS"
 #changelog="https://gitlab.gnome.org/GNOME/gdm/-/raw/main/NEWS"
-distfiles="${GNOME_SITE}/gdm/${version%.*}/gdm-${version}.tar.xz"
-checksum=68266b3abe7d28fc469d0067aac9c5dabb0ca7952cc1f7c238a04951f3dc5b0d
+distfiles="${GNOME_SITE}/gdm/${version:0:2}/gdm-${version}.tar.xz"
```
Comment (oreo639) [inline srcpkgs/gdm/template:30]: Use `${version%%.*}`

**Comment by oreo639** [comment]
Can you pull the following gdm patch: https://gitlab.gnome.org/GNOME/gdm/-/commit/08ef19e848176e58511f1a66992b5a26cb3c51a2.patch (name it `gcc14.patch`)

**Comment by oreo639** [comment]
Can you pull the following xdg-desktop-portals-gnome patch: https://gitlab.gnome.org/GNOME/xdg-desktop-portal-gnome/-/commit/5fb3c0c23ccd76c6f8239bbb66c3fb1230208d5e (name it `gcc14.patch`)

File: srcpkgs/gnome-core/template [hunk 7]
```diff
@@ -38,13 +38,13 @@ depends="
...
  gsettings-desktop-schemas>=${version}
- gvfs>=1.50.0
+ gvfs>=1.52.2
  mutter>=${version}
  nautilus>=${version}
- sushi>=43.0
- tracker>=3.5.0
- yelp>=42.0
+ sushi>=${version}
+ tracker>=3.6.2
```
Comment (oreo639) [inline srcpkgs/gnome-core/template:47]: There is no tracker 3.6.2.

File: srcpkgs/gdm/template (hunk 6 above)
Comment (chrysos349) [reply srcpkgs/gdm/template:30]: fixed

File: srcpkgs/gnome-core/template (hunk 7 above)
Comment (chrysos349) [reply srcpkgs/gnome-core/template]: i wonder how it passed the check.

**Comment by chrysos349** [comment]
- added gcc14-compat patches for `gdm` and `xdg-desktop-portal-gnome`.
- move gnome-45 to this pr.
- rename this pr to gnome-45.0

File: srcpkgs/gnome/template [hunk 8]
```diff
@@ -1,99 +1,104 @@
...
 revision=1
 build_style=meta
 short_desc="GNOME meta-package for Void Linux"
 maintainer="Orphaned <orphan@voidlinux.org>"
 license="GPL-2.0-only, LGPL-2.0-only"
 homepage="https://www.gnome.org"
 
+# loupe replaces eog since gnome-45.0
+# snapshot replaces cheese since gnome-45.0
+
```
Comment (oreo639) [inline srcpkgs/gnome/template:13]: Remove these comments

Comment (chrysos349) [reply srcpkgs/gnome/template]: removed

File: srcpkgs/gdm/template [hunk 9]
```diff
@@ -11,23 +11,24 @@ configure_args="
  -Dsystemd-journal=false -Dinitial-vt=7 -Dwayland-support=true
  -Dselinux=disabled -Dlibaudit=disabled -Dgdm-xsession=true
  -Ddefault-path=/usr/local/sbin:/usr/local/bin:/usr/bin:/usr/sbin:/sbin:/bin
- -Dsystemdsystemunitdir=/usr/lib/systemd/system
- -Dsystemduserunitdir=/usr/lib/systemd/user"
+ -Dsystemdsystemunitdir=no
+ -Dsystemduserunitdir=no
```
Comment (oreo639) [inline srcpkgs/gdm/template:15]: Import the following patch and fix the paths.
https://gist.github.com/oreo639/bde1b7b04a7f1a5d8b1fbc525429ac4c

(That patch has been submitted upstream btw: https://gitlab.gnome.org/GNOME/gdm/-/merge_requests/247)

File: srcpkgs/gnome/template [hunk 10]
```diff
@@ -11,89 +11,91 @@ homepage="https://www.gnome.org"
...
+ power-profiles-daemon>=0.13
+ simple-scan>=44.0
+ snapshot>=${version}
  totem>=43"
 
 _apps_depends="
  Endeavour>=43
  bijiben>=40.1
- dconf-editor>=3.38.3
+ cheese>=${version}
```
Comment (oreo639) [inline srcpkgs/gnome/template:45]: There is no cheese 45. Also, in this instance, cheese can be removed.

File: srcpkgs/gnome/template [hunk 11]
```diff
@@ -11,89 +11,91 @@ homepage="https://www.gnome.org"
...
+ gnome-connections>=44.0
+ gnome-console>=${version}
  gnome-dictionary>=40.0
  gnome-disk-utility>=${version}
- gnome-font-viewer>=${version}
+ gnome-font-viewer>=44.0
  gnome-maps>=${version}
  gnome-music>=${version}
  gnome-nettool>=42
  gnome-photos>=${version}
```
Comment (oreo639) [inline srcpkgs/gnome/template:67]: There is no gnome-photos 45. This should be `gnome-photos>=44.0`.

File: srcpkgs/gdm/template (hunk 9 above)
Comment (chrysos349) [reply srcpkgs/gdm/template]: fixed

**Comment by oreo639** [comment]
I didn't realize, but the tracker-miners update was bad. (seccomp errors with "Bad system call")

**Comment by oreo639** [comment]
Sorry about that, the tracker-miners seccomp failure is related to being built with new glibc, not this update specifically (seems to have been fixed upstream).

**Pull request merged**